python/ml4ir/base/pipeline.py

```diff
@@ -236,6 +212,11 @@ def get_relevance_dataset(self, preprocessing_keys_to_fns={}) -> RelevanceDatase
 
         NOTE: Override this method to create custom dataset objects
         """
+
+        if 'preprocessing_info' in self.feature_config.get_label():
```

Comment:
You should move this to `ml4ir/applications/ranking/pipeline.py`. You can use `super().get_relevance_dataset(...)` method to do this.

Reply:
I'm not sure that I follow. I added these lines to get_relevance_dataset(...)

Reply:
@lastmansleeping can you elaborate a little bit more.

Reply:
Since converting graded relevance to clicks is only a Ranking specific operation, it makes sense to keep it out of the `base/pipeline.py` and move it to `applications/ranking/pipeline.py`

Reply:
In fact, you don't have to do this explicitly. Let's quickly sync offline about this. ml4ir should automatically apply it if you add it to `PreProcessingMap` in `ml4ir/base/features/preprocessing.py`